src/d/llvm/expression.d

```diff
@@ -353,13 +353,19 @@ struct ExpressionGen {
 			
 			case Minus :
 				import d.llvm.type;
-				return LLVMBuildSub(
-					builder,
-					LLVMConstInt(TypeGen(pass.pass).visit(e.type), 0, true),
-					visit(e.expr),
-					"",
-				);
-			
+				import d.common.builtintype : isFloat;
+				if (e.expr.type.builtin.isFloat())
```

Comment:
You do not want to be doing this, because the type system used here retain information such as alias, so this might not be a builtin.